### DISCUSSION TITLE
🐛 firebase_messaging: ^9.1.1 Default Background Notification Bug

### DISCUSSION THREAD

**Discussion by Amanullahgit** [firebase/flutterfire#5704] [Q&A] [answered]
## Bug report

**Describe the bug**
when the app is in background and notification received, this function will be called.
` FirebaseMessaging.onBackgroundMessage(_firebaseMessagingBackgroundHandler);`
```
Future<void> _firebaseMessagingBackgroundHandler(RemoteMessage message) async {
  print('Handling a background message ${message.messageId}');
//tried to apply my own notification style. it worked but i get 2 notifications now. one is default one...
}
```
I am trying to setup my own notification with proper title, body, and styling. but i get a default notification which is having title and body of the notification sent from firebase. 

**How to disable or modify default notification sent by on call of backgroundHandler from FirebaseMessaging ?**

**Comment by Salakar** [comment, marked as answer]
To avoid default notification behaviour; don't send a notification payload - send a data only message (which you can't do directly on the Firebase Console UI/Website) via the server SDKS, see https://firebase.flutter.dev/docs/messaging/server-integration

Send all the data you need to construct your own notification on the device (this is called a local notification) as keys in the `data` object of your FCM message. Then in your `onBackgroundMessage` when the data message comes through build your own notification using the data you sent by using a library such as [flutter_local_notifications](https://pub.dev/packages/flutter_local_notifications) to build and show a local notification.

**Reply by nivisi** [reply]
@Salakar hey! I know it's been a while, sorry for reviving this thread 😀 The proposed solution doesn't work on iOS when the app is killed. The handler is not executed for data-only messages. Is there any way to handle this?

**Reply by rohit-suthar** [reply]
@Salakar ios do not trigger the handler for data-only messages if the app is terminated (Swiped away from app switcher). Is there any way to display our own custom notification with ```notification + data``` payload?
